packages/dx-chart-core/src/plugins/series/computeds.js

```diff
@@ -9,137 +9,146 @@ import {
+  argumentField,
+  valueField,
 ) => {
+  const width = Math.max.apply(null, xScale.range());
+  const height = Math.max.apply(null, yScale.range());
   const radius = Math.min(width, height) / 2;
   const pieData = pie().value(d => d[valueField])(data);
 
-  return pieData.map(d => ({
-    d: arc().innerRadius(radius * innerRadius)
-      .outerRadius(radius * outerRadius || radius)
-      .startAngle(d.startAngle)
-      .endAngle(d.endAngle)(),
-    value: d.value,
+  return pieData.map(({
+    startAngle, endAngle, value, data: itemData,
+  }) => ({
+    d: arc()
+      .innerRadius(0)
+      .outerRadius(radius)
+      .startAngle(startAngle)
+      .endAngle(endAngle)(),
+    value,
+    data: itemData,
+    id: itemData[argumentField],
+    x: width / 2,
+    y: height / 2,
   }));
 };
 
 export const coordinates = (
   data,
-  scales,
+  { xScale, yScale },
   argumentField,
   valueField,
   name,
-) => computeLinePath(data, scales, argumentField, valueField, name);
+) => data.reduce((result, dataItem) => {
+  if (dataItem[argumentField] !== undefined && dataItem[valueField] !== undefined) {
+    return [...result, {
+      x: xScale(dataItem[argumentField]),
+      y: yScale(dataItem[`${valueField}-${name}-stack`][1]),
+      y1: yScale(dataItem[`${valueField}-${name}-stack`][0]),
+      width: xScale.bandwidth ? xScale.bandwidth() : 0,
+      id: dataItem[argumentField],
+      value: dataItem[valueField],
+    }];
+  }
+  return result;
+}, []);
 
-export const findSeriesByName = (name, series) => series.find(
-  seriesItem => seriesItem.uniqueName === name,
-);
+export const barCoordinates = (
+  data,
+  { xScale, yScale },
+  argumentField,
+  valueField,
+  name,
+  stack,
+  stacks = [undefined],
+  { barWidth = 0.9 },
+) => {
+  const rawCoordinates = coordinates(
+    data,
+    { xScale, yScale },
+    argumentField,
+    valueField,
+    name,
+  );
+  const bandwidth = xScale.bandwidth ? xScale.bandwidth() : 0;
+  const x0Scale = createScale(
+    {
+      type: 'band',
+      domain: stacks,
+    },
+    bandwidth,
+    bandwidth,
+    1 - barWidth,
+  );
+  return rawCoordinates.map(item => ({
+    ...item,
+    width: x0Scale.bandwidth(),
+    x: item.x + x0Scale(stack),
+  }));
+};
 
+export const findSeriesByName = (
+  name, series,
+) => series.find(seriesItem => seriesItem.symbolName === name);
 
-export const lineAttributes = (
-  type,
-  scales,
-) => ({
-  path: getGenerator(type),
-  x: scales.xScale.bandwidth ? scales.xScale.bandwidth() / 2 : 0,
-  y: 0,
+export const dBar = ({
+  x, y, y1, width,
+}) => ({
+  x, y: Math.min(y, y1), width: width || 2, height: Math.abs(y1 - y),
 });
 
-export const pointAttributes = (scales, { size = 7 }) => {
-  const dPoint = symbol().size([size ** 2]).type(symbolCircle)();
-  const offSet = scales.xScale.bandwidth ? scales.xScale.bandwidth() / 2 : 0;
+export const pointAttributes = () => {
+  const dPoint = symbol().size([DEFAULT_POINT_SIZE ** 2]).type(symbolCircle)();
```

Comment:
In the doc of the series the 'point.size' option remains. And it looks like BC, that should be described.